grass/kernel.c

```diff
@@ -35,8 +35,11 @@ void kernel_entry(uint mcause) {
     asm("csrr %0, mepc" : "=r"(proc_set[curr_proc_idx].mepc));
     memcpy(proc_set[curr_proc_idx].saved_register, SAVED_REGISTER_ADDR, SAVED_REGISTER_SIZE);
 
+    grass->print_status = PRINT_KERNEL;
     uint id = mcause & 0x3FF;
     (mcause & (1 << 31))? intr_entry(id) : excp_entry(id);
+    grass->print_status = (curr_pid < GPID_SHELL)? PRINT_KERNEL : PRINT_USER;
+    earth->timer_reset(core_in_kernel);
```

Comment:
Hmm, why adding a timer reset here?

Reply:
I found a subtle issue with kernel servers, as they don't call `proc_yield`, meaning their timer is not reset and they interrupt infinitely. I wished to always reset the timer before moving back to User Space, but to make the reasoning clearer, the timer reset has been moved to the bottom of `intr_entry`.

Reply:
Got it -- thanks.